### PR TITLE
perf: Optimize SlickGrid handleSelectedRangesChanged by using Set

### PR DESCRIPTION
**Pull request by Vsinghal339-source**
Performance optimisation from O(n square) to O(n)

### CODE REVIEW

**Comment by Vsinghal339-source** [comment]
@ghiscoding Please review

File: packages/common/src/core/slickGrid.ts [hunk 1]
```diff
@@ -2917,8 +2917,12 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
...
     if (this.simpleArrayEquals(previousSelectedRows, this.selectedRows)) {
       const caller = ne?.detail?.caller ?? 'click';
-      const newSelectedAdditions = this.getSelectedRows().filter((i) => previousSelectedRows.indexOf(i) < 0);
-      const newSelectedDeletions = previousSelectedRows.filter((i) => this.getSelectedRows().indexOf(i) < 0);
+      // Use Set for faster performance
+      const selectedRowsSet = new Set(this.getSelectedRows());
+      const previousSelectedRowsSet = new Set(previousSelectedRows);
+
+      const newSelectedAdditions = [...selectedRowsSet].filter(i => !previousSelectedRowsSet.has(i));
+      const newSelectedDeletions = [...previousSelectedRowsSet].filter(i => !selectedRowsSet.has(i));
```
Comment (ghiscoding) [inline packages/common/src/core/slickGrid.ts:2925]: @Vsinghal339-source  why are you making a copy of each array `[...array]`, is it really necessary? I mean doing this is an extra step to duplicate each array. So I wonder why you are doing this step?

**Comment by Vsinghal339-source** [comment]
@ghiscoding Done

**Comment by ghiscoding** [comment]
Thanks again for another great contribution

**Comment by ghiscoding** [comment]
All released as v5.7.0

**Pull request merged**